command/connect/envoy/envoy.go

```diff
@@ -612,6 +613,16 @@ func (c *cmd) generateConfig() ([]byte, error) {
 
 	var bsCfg BootstrapConfig
 
+	// Make a call to an arbitrary ACL endpoint. If we get back an ErrNotFound
+	// (meaning ACLs are enabled) check that the token is not empty.
+	if _, _, err := c.client.ACL().TokenReadSelf(
+		&api.QueryOptions{Token: args.Token},
+	); acl.IsErrNotFound(err) {
+		if args.Token == "" {
+			c.UI.Warn("ACL is enabled but no token was provided.")
```

Comment:
"ACL is enabled but no token was provided." sounds good and describes the situation. I wonder if we can describe the potential cost& suggest fix as well. Maybe something like **"ACL is enabled but no token was provided. Please pass the correct token to Envoy to avoid potential access denial and communication failure."**? 
*Should it be "pass **a** token" or "pass **the** token"? I wish to be clear that they need the right token for the ACL they set up and make them think "a token? any token's fine?" or "the token? which token?". At the same time, I don't want to stress user with too many details like another doc to read&learn @jkirschner-hashicorp wdyt?
*Not sure if I described the consequences precisely, please feel free to edit. If possible, shorten "access denial and communication failure" this part would be cool.

Reply:
Maybe "ACL is enabled but no token was provided. To avoid potential access denial and communication failure, please pass the authorized token to Envoy."